src/parser/smt2/smt2.cpp

```diff
@@ -1825,6 +1824,18 @@ Expr Smt2::applyParseOp(ParseOp& p, std::vector<Expr>& args)
     const Datatype& dt = ((DatatypeType)t).getDatatype();
     return em->mkExpr(kind::APPLY_SELECTOR, dt[0][n].getSelector(), args);
   }
+  else if (!em->getOptions().getUfHo()
+           && (kind == kind::EQUAL || kind == kind::DISTINCT))
```

Comment:
Will this catch everything? What about functions as arguments to other functions?  Or as arguments to other interpreted functions?

Reply:
I think those will be caught by the type checker. This code catches the case for equality, which is parametric so it's fine in terms of types.